### PR TITLE
Support Oryx 10GigE camera

### LINKED ISSUE #3: Support Oryx 10GigE camera
Modify the spinnaker drivers such that basic usage of an Oryx 10GigE is supported.

Specifically we are targeting the [ORX-10GS-51S5M-C model](https://www.flir.com/products/oryx-10gige/?model=ORX-10GS-51S5M-C&vertical=machine+vision&segment=iis).

Basic usage is defined as the supported behaviors implied by [the tests in the acquire-driver-egrabber repo](https://github.com/acquire-project/acquire-driver-egrabber/tree/d9724542a8fadb053fa03d1cfc17e637a95fec04/tests).

Dependent on https://github.com/acquire-project/acquire-driver-spinnaker/issues/2 for an initial spinnaker driver implementation.

### PR DESCRIPTION
**Pull request by andy-sweet**
This adds support for an Oryx 10GigE (ORX-10GS-51S5M-C) camera.

The underlying implementation is shared with the previously support Blackfly camera. The advantage of this approach is that there is less code to maintain and supporting other Spinnaker based cameras could be little to no work.

The disadvantage of this approach is that the implementation related to triggers is complicated. Alternatively, we could consider completely splitting at least part of the implementation and using different parts of it explicitly based on the camera model.

This also adds tests for the Oryx camera that mirror those for the Blackfly. A test for `Camera::get_meta` was also added for both cameras. I also added a ctest setup fixture to ensure that the packet size is set to 1000B for the Oryx camera.

Closes #3

### CODE REVIEW

**Comment by andy-sweet** [comment]
All tests are passing locally, but only the Blackfly tests are passing on CI as there is no Oryx camera connected to a runner right now.

**Comment by andy-sweet** [comment]
Both cameras are connceted to Ghanima and I added a setup fixture to force the packet size to be 1000B. All tests pass, except the first blackfly specific test, which has failed twice in a row with a similar error that likely occurs when initializing the camera. The latest error is the following.

```
 Exception: Spinnaker: Error reading from Device [SN:23077028], at register address: F0F01120. Please try reconnecting the device. [-1011]
``` 

Note that all subsequent blackly tests pass and all oryx tests pass.

I don't believe I've seen this in previous runs, so it's possible that the fixture is causing it somehow.

UPDATE: I temporarily disabled the fixture and saw a similar error, though it did not occur for the first blackfly test. So this is probably not directly caused by the fixture.

File: src/spinnaker.cpp [hunk 1]
```diff
@@ -724,29 +820,80 @@ SpinnakerCamera::query_pixel_type_capabilities_(
...
+                    meta->triggers.acquisition_start.output |= (1ULL << i);
+                }
+                if (is_enum_name(camera_->LineSource, genicam_frame_start)) {
+                    meta->triggers.frame_start.output |= (1ULL << i);
+                }
+                if (is_enum_name(camera_->LineSource,
+                                 genicam_exposure_active)) {
+                    meta->triggers.exposure.output |= (1ULL << i);
+                }
+                // TODO: what about trigger_wait?
```
Comment (aliddell) [inline src/spinnaker.cpp]: 👀

**Review by aliddell — APPROVED** [review]
I don't see any real issues here.

File: src/spinnaker.cpp (hunk 1 above)
Comment (nclack) [reply src/spinnaker.cpp]: "trigger_wait" is an output line that should go high when the camera is awaiting a trigger. I confused myself and had to look up the name.

@andy-sweet It looks like you might have handled this via "FrameTriggerWait"? Is this todo resolved?

File: src/spinnaker.cpp [hunk 2]
```diff
@@ -227,7 +303,7 @@ struct SpinnakerCamera final : private Camera
 
     void set(struct CameraProperties* properties);
     void get(struct CameraProperties* properties);
-    void get_meta(struct CameraPropertyMetadata* meta) const;
+    void get_meta(struct CameraPropertyMetadata* meta);
```
Comment (nclack) [inline src/spinnaker.cpp]: why did this one go non-const?

**Review by nclack — COMMENTED** [review]
This looks good to me.

There's a TODO I'm not sure about and I asked one other question, but otherwise this is gtg.

File: src/spinnaker.cpp (hunk 2 above)
Comment (andy-sweet) [reply src/spinnaker.cpp]: Oops. I did that temporarily for reasons, but then those reasons no longer applied and I forgot to update it back. Added a commit to add the `const` back and will push it soon.

File: src/spinnaker.cpp (hunk 1 above)
Comment (andy-sweet) [reply src/spinnaker.cpp]: This method is trying to populate the trigger capabilities by filling out `CameraPropertiesTriggerMetadata`.

The problem is that only contains input/output bit-masks for `acquisition_start, exposure, frame_start`, but not `trigger_wait`: https://github.com/acquire-project/acquire-core-libs/blob/40e71b2af97f8827be5d88352414365193145ed3/src/acquire-device-properties/device/props/camera.h#L76

I.e. users can configure the `trigger_wait` output line/trigger in the properties and I pass that through to spinnaker, but there's no way to express the capability in the properties metadata.

My plan here is to remove the TODO and create an issue on acquire-core-libs. Does that make sense?

**Comment by andy-sweet** [comment]
Thanks for the reviews!

File: src/spinnaker.cpp (hunk 1 above)
Comment (andy-sweet) [reply src/spinnaker.cpp]: Resolved in person. Will remove this and create issues.

**Pull request merged**